contracts/StableJoeStaking.sol

```diff
@@ -0,0 +1,181 @@
+
+    /// @notice The reward token.
+    IERC20Upgradeable public rewardToken;
+
+    /// @notice Info of each user that stakes moJoe.
+    mapping(address => UserInfo) public userInfo;
+
+    /// @notice Emitted when a user deposits its moJOE
+    event Deposit(address indexed user, uint256 amount);
+
+    /// @notice Emitted when a user withdraws its moJOE
+    event Withdraw(address indexed user, uint256 amount);
+
+    /// @notice Emitted when a user emergency withdraws its moJOE
+    event EmergencyWithdraw(address indexed user, uint256 amount);
+
+    /**
+     * @notice Initialize a new StableJoeStaking contract.
+     * @dev This contract needs to receive an ERC20 `_rewardToken` in order to distribute them
+     * (with JoeMakerV4 in our case)
+     * @param _rewardToken The address of the ERC-20 reward token
+     * @param _moJoe The address of the ERC-20 moJOE token, traderjoe's staking token.
+     */
+    function initialize(IERC20Upgradeable _rewardToken, IERC20Upgradeable _moJoe) public initializer {
+        __Ownable_init();
+
+        rewardToken = _rewardToken;
+        moJoe = _moJoe;
+    }
+
+    /**
+     * @notice View function to see pending reward token on frontend.
+     * @param _user The address of the user
+     * @return `_user`'s pending reward token
+     */
+    function pendingToken(address _user) external view returns (uint256) {
+        UserInfo storage user = userInfo[_user];
+        uint256 lpSupply = moJoe.balanceOf(address(this));
+        uint256 _accTokenPerShare = accTokenPerShare;
+
+        uint256 rewardTokenBalance = rewardToken.balanceOf(address(this));
+        if (rewardTokenBalance != lastTokenBalance && lpSupply != 0) {
+            uint256 rewardTokenAdded = rewardTokenBalance.sub(lastTokenBalance);
+            _accTokenPerShare = _accTokenPerShare.add(rewardTokenAdded.mul(1e12).div(lpSupply));
+        }
+        return user.amount.mul(_accTokenPerShare).div(1e12).sub(user.rewardDebt);
+    }
+
+    /**
+     * @notice Update reward variables to be up-to-date.
+     * @dev Needs to be called before any deposit or withdrawal.
+     */
+    function updatePool() public {
+        uint256 rewardTokenBalance = rewardToken.balanceOf(address(this));
+        if (rewardTokenBalance == lastTokenBalance) {
+            return;
+        }
+
+        uint256 lpSupply = moJoe.balanceOf(address(this));
+        if (lpSupply == 0) {
+            lastTokenBalance = rewardTokenBalance;
+            return;
+        }
+
+        uint256 rewardTokenAdded = rewardTokenBalance.sub(lastTokenBalance);
+        accTokenPerShare = accTokenPerShare.add(rewardTokenAdded.mul(1e12).div(lpSupply));
+        lastTokenBalance = rewardTokenBalance;
+    }
+
+    /**
+     * @notice Deposit moJoe to sJoe for reward token allocation.
+     * @param _amount The amount of moJOE to deposit
+     */
+    function deposit(uint256 _amount) public {
+        UserInfo storage user = userInfo[msg.sender];
+
+        updatePool();
```

Comment:
nit: move `updatePool()` call to be first line of this function to make it explicit it needs to be called before proceeding with any logic